src/Nethermind/Nethermind.Network.Discovery/Serializers/FindNodeMsgSerializer.cs

```diff
@@ -1,53 +1,65 @@
+//
 //  The Nethermind library is free software: you can redistribute it and/or modify
 //  it under the terms of the GNU Lesser General Public License as published by
 //  the Free Software Foundation, either version 3 of the License, or
 //  (at your option) any later version.
-// 
+//
 //  The Nethermind library is distributed in the hope that it will be useful,
 //  but WITHOUT ANY WARRANTY; without even the implied warranty of
 //  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 //  GNU Lesser General Public License for more details.
-// 
+//
 //  You should have received a copy of the GNU Lesser General Public License
 //  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
 
+using DotNetty.Buffers;
 using Nethermind.Core.Crypto;
 using Nethermind.Crypto;
 using Nethermind.Network.Discovery.Messages;
+using Nethermind.Network.P2P;
 using Nethermind.Serialization.Rlp;
 
 namespace Nethermind.Network.Discovery.Serializers;
 
-public class FindNodeMsgSerializer : DiscoveryMsgSerializerBase, IMessageSerializer<FindNodeMsg>
+public class FindNodeMsgSerializer : DiscoveryMsgSerializerBase, IZeroMessageSerializer<FindNodeMsg>
 {
     public FindNodeMsgSerializer(IEcdsa ecdsa, IPrivateKeyGenerator nodeKey, INodeIdResolver nodeIdResolver)
         : base(ecdsa, nodeKey, nodeIdResolver) { }
 
-    public byte[] Serialize(FindNodeMsg msg)
+    public void Serialize(IByteBuffer byteBuffer, FindNodeMsg msg)
     {
-        byte[] data = Rlp.Encode(
-            Rlp.Encode(msg.SearchedNodeId),
-            //verify if encoding is correct
-            Rlp.Encode(msg.ExpirationTime)
-        ).Bytes;
-
-        byte[] serializedMsg = Serialize((byte) msg.MsgType, data);
-        return serializedMsg;
+        int length = GetLength(msg, out int contentLength);
+
+        RlpStream stream = new(length);
+
+        stream.StartSequence(contentLength);
+        stream.Encode(msg.SearchedNodeId);
+        stream.Encode(msg.ExpirationTime);
+
+        byte[] serializedMsg = Serialize((byte) msg.MsgType, stream.Data);
+        byteBuffer.EnsureWritable(serializedMsg.Length);
+        byteBuffer.WriteBytes(serializedMsg);
     }
 
-    public FindNodeMsg Deserialize(byte[] msgBytes)
+    public FindNodeMsg Deserialize(IByteBuffer msgBytes)
     {
-        (PublicKey FarPublicKey, byte[] Mdc, byte[] Data) results = PrepareForDeserialization(msgBytes);
-        RlpStream rlpStream = results.Data.AsRlpStream();
-
+        (PublicKey FarPublicKey, byte[] Mdc, IByteBuffer Data) results = PrepareForDeserialization(msgBytes);
+        NettyRlpStream rlpStream = new(results.Data);
         rlpStream.ReadSequenceLength();
         byte[] searchedNodeId = rlpStream.DecodeByteArray();
         long expirationTime = rlpStream.DecodeLong();
 
         FindNodeMsg findNodeMsg = new (results.FarPublicKey, expirationTime, searchedNodeId);
         return findNodeMsg;
     }
+
+    private int GetLength(FindNodeMsg msg, out int contentLength)
```

Comment:
Maybe
_private (int legth, int contentLength) GetLength(FindNodeMsg msg)_
?

Reply:
it was like this for GetLength in some serializers. So I kept the same interface.
I was thinking, do we want to standardize this GetLength interface for all the serializers? So that we can query the length before allocating the IByteBuffer?